File: snippets/rust/src/configuration.rs
use anyhow::Result;
use breez_sdk_liquid::prelude::*;


fn configure_asset_metadata() -> Result<Config> {
    // ANCHOR: configure-asset-metadata
    // Create the default config
    let mut config = LiquidSdk::default_config(
        LiquidNetwork::Mainnet,
        Some("<your-Breez-API-key>".to_string()),
    )?;

    // Configure asset metadata. Setting the optional fiat ID will enable
    // paying fees using the asset (if available).
    config.asset_metadata = Some(vec![
        AssetMetadata {
            asset_id: "18729918ab4bca843656f08d4dd877bed6641fbd596a0a963abbf199cfeb3cec".to_string(),
            name: "PEGx EUR".to_string(),
            ticker: "EURx".to_string(),
            precision: 8,
            fiat_id: Some("EUR".to_string()),
        },
    ]);
    // ANCHOR_END: configure-asset-metadata
    Ok(config)
}

fn configure_parsers() -> Result<Config> {
    // ANCHOR: configure-external-parser
    // Create the default config
    let mut config = LiquidSdk::default_config(
        LiquidNetwork::Mainnet,
        Some("<your-Breez-API-key>".to_string()),
    )?;

    // Configure external parsers
    config.external_input_parsers = Some(vec![
        ExternalInputParser {
            provider_id: "provider_a".to_string(),
            input_regex: "^provider_a".to_string(),
            parser_url: "https://parser-domain.com/parser?input=<input>".to_string(),
        },
        ExternalInputParser {
            provider_id: "provider_b".to_string(),
            input_regex: "^provider_b".to_string(),
            parser_url: "https://parser-domain.com/parser?input=<input>".to_string(),
        },
    ]);
    // ANCHOR_END: configure-external-parser
    Ok(config)
}

fn configure_magic_routing_hints() -> Result<Config> {
    // ANCHOR: configure-magic-routing-hints
    // Create the default config
    let mut config = LiquidSdk::default_config(
        LiquidNetwork::Mainnet,
        Some("<your-Breez-API-key>".to_string()),
    )?;

    // Configure magic routing hints
    config.use_magic_routing_hints = false;
    // ANCHOR_END: configure-magic-routing-hints
    Ok(config)
}

File: snippets/python/src/configuration.py
from breez_sdk_liquid import AssetMetadata, default_config, LiquidNetwork, ExternalInputParser


def configure_asset_metadata():
    # ANCHOR: configure-asset-metadata
    # Create the default config
    config = default_config(
        network=LiquidNetwork.MAINNET,
        breez_api_key="<your-Breez-API-key>"
    )

    # Configure asset metadata. Setting the optional fiat ID will enable
    # paying fees using the asset (if available).
    config.asset_metadata = [
        AssetMetadata(
            asset_id="18729918ab4bca843656f08d4dd877bed6641fbd596a0a963abbf199cfeb3cec",
            name="PEGx EUR",
            ticker="EURx",
            precision=8,
            fiat_id="EUR"
        )
    ]
    # ANCHOR_END: configure-asset-metadata

def configure_parsers():
    # ANCHOR: configure-external-parser
    # Create the default config
    config = default_config(
        network=LiquidNetwork.MAINNET,
        breez_api_key="<your-Breez-API-key>"
    )

    # Configure external parsers
    config.external_input_parsers = [
        ExternalInputParser(
            provider_id="provider_a",
            input_regex="^provider_a",
            parser_url="https://parser-domain.com/parser?input=<input>"
        ),
        ExternalInputParser(
            provider_id="provider_b",
            input_regex="^provider_b",
            parser_url="https://parser-domain.com/parser?input=<input>"
        )
    ]
    # ANCHOR_END: configure-external-parser

def configure_magic_routing_hints():
    # ANCHOR: configure-magic-routing-hints
    # Create the default config
    config = default_config(
        network=LiquidNetwork.MAINNET,
        breez_api_key="<your-Breez-API-key>"
    )

    # Configure magic routing hints
    config.use_magic_routing_hints = False
    # ANCHOR_END: configure-magic-routing-hints
